src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
+   buffer = nn_allocmsg(data_size, 0);
+   auto error = nn_errno ();
+   if (error == ENOMEM) {
+      throw std::runtime_error("not enough memory to allocate NanoMsg: " +
+                               std::string(nn_strerror(errno)));
+   }
+   std::memcpy(buffer, &data[0], data_size);
```

Comment:
maybe ```bufferSize``` is a better name. When I read this I got confused and had to go back and check between ```data.size()``` and ```data_size```